### PR TITLE
fix: Fix issues with update throttler

### LINKED ISSUE #3398: [BUG] Instruments update at half rate when in separate window
**Mod Version**
master ToT (Tip-of-Tree)

**Describe the bug**
Instruments update at half rate when in separate window

**To Reproduce**
1. Hover over ND and press Alt Gr + Left Mouse click

**Expected behavior**
Smooth update rate

**Actual behavior**
Update rate is half of the expected rate (only ~2fps)

**Additional context**

Was this working before/when did the issue start occurring?
Regression from the throttling PRs

Is this a problem in the vanilla unmodded game? No

Discord username (if different from GitHub): JoJo#7105

### PR DESCRIPTION
**Pull request by XSJoJo**
Fixes #3398

## Summary of Changes
- Ensure that _deltaTime that gets passed into the throttler is accumulated in case of
lower instrument refresh rate setting.
- Update map more frequently when turning.

## Testing instructions
- Load any flight plan.
- Do some turns and fly along
- Observe how flight plan update is smoother.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A320_Neo_MFD.js [hunk 1]
```diff
@@ -629,17 +631,28 @@ class A320_Neo_MFD_Map extends MapInstrumentElement {
...
                 this.lastMcduCurrentFplnWaypointIndex = mcduCurrentFplnWaypointIndex;
                 forceUpdate = true;
             }
         }
+
+        // When the plane is turning, update map faster.
+        const heading = Simplane.getHeadingMagnetic();
+        if (Math.abs(heading - this.lastHeading) > this.headingDiffForceUpdateThreshold) {
+            forceUpdate = true;
+
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A320_Neo_MFD.js:646]:
```suggestion
```

**Review by Benjozork — APPROVED** [review]
LGTM

**Comment by omarsmak** [comment]
Is this PR testable already?

**Comment by XSJoJo** [comment]
> Is this PR ready for testing?

Yes.

**Comment by GlitchedYT** [comment]
Category: QA Tester
Name: Glitched#0740
Date of testing: Current date 02/17/2021
Version of the sim: 1.13.16.0
PR Tested: #3563
Tier of testing: 2
Changes to observe: 
- Ensure that _deltaTime that gets passed into the throttler is accumulated in case of
  lower instrument refresh rate setting.
- Update map more frequently when turning.
(Insert the Summary of changes, in order to remind what is to test)

Testing technique: Flew patterns around KMIA at low, medium, and high altitudes to see if the ND lagged at all
Steps you've made to test the Pull Request:
1. Spawned on runway an KMIA
2. Loaded flight from KMIA - KEYW
3. T/O and observed ND for any lagging/stuttering
4. Did circle's at low, medium, and high altitudes at different speeds
5.

Flight notes (When done a full flight): 
- Refresh rate on LOW

Media: Insert screenshots, recordings, or anything relevant to the review here.
https://streamable.com/odawgo

Issues (If Any): 
- Captain ND is definitely noticeably improved but could be even smoother if it doesn't affect fps much.
- FO ND still seems laggy when making sharp turns 
Steps to achieve the issue

Overall Rating: To be improved
Conclusions of the testing: Make a small descriptive to areas where it needs to be improved and recommendations.

**Comment by XSJoJo** [comment]
> Issues (If Any):
> 
> * Captain ND is definitely noticeably improved but could be even smoother if it doesn't affect fps much.
> * FO ND still seems laggy when making sharp turns
>   Steps to achieve the issue
> 
> Overall Rating: To be improved
> Conclusions of the testing: Make a small descriptive to areas where it needs to be improved and recommendations.

Talked offline - this is probably the right balance between Sim FPS and ND FPS at the moment, given how expensive flight plan drawing is. Once the new flight plan drawing algorithm is in, I'm happy to revisit this.

**Comment by tracernz** [comment]
>  Once the new flight plan drawing algorithm is in, I'm happy to revisit this.

Is somebody working on that? I wonder what nanovg performance is like. Could be worthwhile for the ND if it's decent.

**Comment by XSJoJo** [comment]
> > Once the new flight plan drawing algorithm is in, I'm happy to revisit this.
> 
> Is somebody working on that? I wonder what nanovg performance is like. Could be worthwhile for the ND if it's decent.

@beheh is working on flight plan drawing that should be more optimized.

**Comment by beheh** [comment]
Hm, for clarification, not directly. As part of my LNAV work I need to rework the ND track lines as they're driven by the new guidance system, rather than the ND doing it's own bezels. So I'm only really changing the source of the SVG paths, not the rendering directly. In future we will most likely still need someone doing a significant rework or performance optimizations.

I also don't think my work should block any optimizations, I can easily adapt the SVG generation to any new ND code as the heavy lifting is done by a (separate) guidance system itself.

**Comment by XSJoJo** [comment]
> Hm, for clarification, not directly. As part of my LNAV work I need to rework the ND track lines as they're driven by the new guidance system, rather than the ND doing it's own bezels. So I'm only really changing the source of the SVG paths, not the rendering directly. In future we will most likely still need someone doing a significant rework or performance optimizations.
> 
> I also don't think my work should block any optimizations, I can easily adapt the SVG generation to any new ND code as the heavy lifting is done by a (separate) guidance system itself.

Right, I suspect though that the code to create the SVG is the slow part, and you're replacing that. But I'll need to measure first to be sure...

**Comment by beheh** [comment]
This seems to be a rather minor change, and we've seen one QA test for it. Any objection to merging this?

**Pull request merged**